File: projects/tox4j/src/main/java/im/tox/tox4j/core/callbacks/FriendLosslessPacketCallback.scala
package im.tox.tox4j.core.callbacks

import im.tox.tox4j.core.data.{ToxFriendNumber, ToxLosslessPacket}
import org.jetbrains.annotations.NotNull

/**
 * This event is triggered when a custom lossless packet arrives from a friend.
 */
trait FriendLosslessPacketCallback[ToxCoreState] {
  /**
   * @param friendNumber The friend number of the friend who sent a lossless packet.
   * @param data A byte array containing the received packet data. The first byte is the packet id.
   */
  def friendLosslessPacket(
    friendNumber: ToxFriendNumber, @NotNull data: ToxLosslessPacket
  )(state: ToxCoreState): ToxCoreState = state
}

File: lib/src/commonMain/kotlin/im/tox/tox4j/core/callbacks/FriendLosslessPacketCallback.kt
package im.tox.tox4j.core.callbacks

import im.tox.tox4j.core.data.ToxFriendNumber
import im.tox.tox4j.core.data.ToxLosslessPacket

/** This event is triggered when a lossless packet is received from a friend. */
interface FriendLosslessPacketCallback<ToxCoreState> {
    /**
     * @param friendNumber The friend number of the friend who sent the packet.
     * @param data A byte array containing the received packet data.
     */
    fun friendLosslessPacket(
        friendNumber: ToxFriendNumber,
        data: ToxLosslessPacket,
        state: ToxCoreState,
    ): ToxCoreState = state
}
